src/ray/common/ray_object.h

```diff
@@ -127,6 +150,9 @@ class RayObject {
   }
 
   std::shared_ptr<Buffer> data_;
+  /// The data factory is used to allocate data from the language frontend.
+  /// Note that, if this is provided, `data_` should be null.
+  std::function<const std::shared_ptr<Buffer>()> data_factory_ = nullptr;
```

Comment:
can we create a different implementation of Buffer (such as JvmManagedBuffer),  instead of data_factory_?

Reply:
The main reason why we need to introduce this data_factory_ is that, if we hold the "data_" in C++, the JVM cannot perform garbage collection on the Java byte array object. Thus, we use this factory to only temporarily get a reference to the Java object when we do "GetData".